types/ember__object/test/proxy.ts

```diff
@@ -0,0 +1,27 @@
+import ObjectProxy from "@ember/object/proxy";
+import { assertType } from "./lib/assert";
+
+interface Foo {
+  fooProp1: string;
+  fooProp2: number;
+}
+
+type AllowUndefined<T> = {
+  [P in keyof T]: T[P] | undefined
+};
```

Comment:
Slightly different (see above comment). `AllowUndefined` does not make the key optional. I can't see a builtin type that does this, but I may have missed it.

Reply:
I'm less concerned about this one since it's in tests, but I believe you could effectively get away with `Partial<T>` here as well. Up to you whether you want to keep this one :)

Reply:
Yep, this one has been removed as well.